src/schema/schema.graphql

```diff
@@ -0,0 +1,67 @@
+scalar Date
+
+type Author {
+    id: ID!
+    name: String!
+    biography: String!
+    bornDate: Date!
+    books: [Book!]
+}
+
+type Book {
+    id: ID!
+    title: String!
+    description: String!
+    publishedDate: Date!
+    authors: [Author!]
+    reviewsAndRating: [BookReviewsAndRating!]
+}
+
+type BookReviewsAndRating {
+    id: ID!
+    bookId: ID!
+    userId: ID!
+    review: String!
+    rating: Int!
+}
+
+type BookPage {
+    books: [Book!]
+    totalPage: Int
+    currentPage: Int
+}
+
+type AuthorPage {
+    authors: [Author!]
+    totalPage: Int
+    currentPage: Int
+}
```

Comment:
Also readup on how cursor based pagination works, would be useful to know and see how to implement that in graphql.